catroid/src/main/java/org/catrobat/catroid/formulaeditor/Formula.java

```diff
@@ -161,7 +164,25 @@ public String interpretString(Sprite sprite) throws InterpretationException {
 			throw new InterpretationException("NaN in interpretString()");
 		}
 
-		return String.valueOf(interpretation);
+		String value = String.valueOf(interpretation);
+		String returnedValue;
+		if (value.contains(".")) {
+			try {
+				String[] splitter = value.split("\\.");
+				int afterDecimalCount = splitter[1].length();
+				String pattern = "#.";
+				for (int i = 0; i < afterDecimalCount; i++) {
+					pattern = pattern.concat("#");
+				}
+				returnedValue = String.valueOf(new DecimalFormat(pattern).format(Double.parseDouble(value)));
+			} catch (Exception e) {
+				Log.e(TAG, e.getMessage());
+				returnedValue = value;
+			}
+		} else {
+			returnedValue = value;
+		}
+		return returnedValue;
```

Comment:
I can't say if I'll review your PR due to time constraints… It's very likely that this will be my only comment on this PR :scream_cat:
However, the trailing zero is a problem addressing multiple components (related tickets). It would be great if you can create a static function for that. It will also be really easy to test.

> Ideally, there should be a single implementation for formatting these numbers…

Reply:
I agree with you, could you suggest the name of the methode :( I still have problem with naming variables and methodes ^_#

Reply:
👍